sonar-db/src/main/java/org/sonar/db/purge/PurgeDao.java

```diff
@@ -209,4 +214,32 @@ public PurgeDao deleteSnapshots(DbSession session, PurgeProfiler profiler, Purge
   private static PurgeMapper mapper(DbSession session) {
     return session.getMapper(PurgeMapper.class);
   }
+
+  private static class DeleteIssueChangesFromIssueKeys implements Function<List<String>, List<Void>> {
+    private final PurgeMapper mapper;
+
+    private DeleteIssueChangesFromIssueKeys(PurgeMapper mapper) {
+      this.mapper = mapper;
+    }
+
+    @Override
+    public List<Void> apply(@Nonnull List<String> input) {
+      mapper.deleteIssueChangesFromIssueKeys(input);
+      return emptyList();
+    }
+  }
+
+  private static class DeleteIssuesFromKeys implements Function<List<String>, List<Void>> {
+    private final PurgeMapper mapper;
+
+    private DeleteIssuesFromKeys(PurgeMapper mapper) {
+      this.mapper = mapper;
+    }
+
+    @Override
+    public List<Void> apply(@Nonnull List<String> input) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this "Nonnull" annotation to honor the overridden method's contract. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2638)